lib/metadataTypes/Interaction.js

```diff
@@ -21,16 +24,265 @@ class Interaction extends MetadataType {
             retrieveDir,
-            `/interaction/v1/interactions${
-                key ? '/key:' + encodeURIComponent(key) : ''
-            }?extras=all`,
+            `/interaction/v1/interactions${singleKey}?extras=${extras}`,
             null,
             null,
             key
         );
     }
+    /**
+     * Delete a metadata item from the specified business unit
+     *
+     * @param {TYPE.BuObject} buObject references credentials
+     * @param {string} key Identifier of item
+     * @param {number} version required version of metadata
+     * @returns {Promise.<boolean>} deletion success status
+     */
+    static async deleteByKey(buObject, key, version) {
+        if (Number.isNaN(version)) {
+            throw new TypeError(
+                'Version is required for deleting interactions to avoid accidental deletion of the wrong item.'
+            );
+        }
+        let singleKey = '';
+        /* eslint-disable unicorn/prefer-ternary */
+        if (key.startsWith('id:')) {
+            // ! allow selecting journeys by ID because that's what users see in the URL
+            singleKey = key.slice(3);
+        } else {
+            // delete by key with specified version does not work, therefore we need to get the ID first
+            const response = await this.client.rest.get(
+                `/interaction/v1/interactions/key:${key}?extras=`
+            );
+            const results = this.parseResponseBody(response, key);
+            singleKey = results[key].id;
+            Util.logger.debug(`Deleting interaction ${key} via its ID ${singleKey}`);
+        }
+        Util.logger.warn(
+            `Deleting Interactions via this command breaks following retrieve-by-key/id requests until you've deployed/created a new draft version! You can get still get the latest available version of your journey by retrieving all interactions on this BU.`
+        );
+        /* eslint-enable unicorn/prefer-ternary */
+        return super.deleteByKeyREST(
+            buObject,
+            '/interaction/v1/interactions/' + singleKey + `?versionNumber=${version}`,
+            key,
+            false
+        );
+    }
+    /**
+     * Updates a single item
+     *
+     * @param {TYPE.MetadataTypeItem} metadata a single item
+     * @returns {Promise} Promise
+     */
+    static update(metadata) {
+        return super.updateREST(metadata, '/interaction/v1/interactions/', true);
+    }
+
+    /**
+     * Creates a single item
+     *
+     * @param {TYPE.MetadataTypeItem} metadata a single item
+     * @returns {Promise} Promise
+     */
+    static create(metadata) {
+        return super.createREST(metadata, '/interaction/v1/interactions/');
+    }
+
+    /**
+     * manages post retrieve steps
+     *
+     * @param {TYPE.MetadataTypeItem} metadata a single query
+     * @returns {TYPE.MetadataTypeItem} Array with one metadata object and one query string
+     */
+    static postRetrieveTasks(metadata) {
```

Comment:
so basically it is only partial given it will only basically work with email?

Reply:
it _works_ with everything but there are lots of IDs that might need to be resolved manually for cross-BU deployments.
As of now, it will allow updating & backing up journeys. I tried adding as many tests as possible for now, but it's certainly still lacking additional postRetrive & preDeploy processing to work as seamlessly as other types do.

I would call this a beta release officially which will be improved over time. Luckily, journeys often use keys already, making custom r__ or c__ fields irrelevant for the most part

Reply:
ensure that there is a real comment to explain limited support like "IMPORTANT: this is for beta release supporting only native external keys OR triggered sends, not all activities are supported"
AND ADD A LOG/WARN to user

Reply:
done :)